DllImportGenerator/Ancillary.Interop/MarshalEx.cs

```diff
@@ -25,5 +25,79 @@ public static void SetHandle(SafeHandle safeHandle, IntPtr handle)
         {            
             typeof(SafeHandle).GetMethod("SetHandle", BindingFlags.NonPublic | BindingFlags.Instance)!.Invoke(safeHandle, new object[] { handle });
         }
+
+        /// <summary>
+        /// Set the last platform invoke error on the thread 
+        /// </summary>
+        public static void SetLastWin32Error(int error)
+        {
+            typeof(Marshal).GetMethod("SetLastWin32Error", BindingFlags.NonPublic | BindingFlags.Static)!.Invoke(null, new object[] { error });
+        }
+
+        /// <summary>
+        /// Get the last system error on the current thread (errno on Unix, GetLastError on Windows)
+        /// </summary>
+        public static unsafe int GetLastSystemError()
+        {
+            // Would be internal call that handles getting the last error for the thread using the PAL
+
+            if (OperatingSystem.IsWindows())
+            {
+                return Kernel32.GetLastError();
+            }
+            else if (OperatingSystem.IsMacOS())
+            {
+                return *libc.__error();
+            }
+            else if (OperatingSystem.IsLinux())
+            {
+                return *libc.__errno_location();
+            }
+
+            throw new NotImplementedException();
```

Comment:
Chose final `else` clause or not - see `SetLastSystemError` below.  I have no opinion.

Reply:
I prefer avoiding the `else` if I can. I only have it in `SetLastSystemError` because those cases don't return and I didn't think it would look any cleaner to make them return. I didn't see much value in making it look the same in this case, so I left this one without a final `else` clause.

If we really want that visual parity, I'll put this one in an `else`.